testkit/src/main/scala/org/bitcoins/testkit/eclair/rpc/EclairRpcTestUtil.scala

```diff
@@ -645,12 +645,14 @@ trait EclairRpcTestUtil extends BitcoinSLogger {
     logger.debug(s"shutting down eclair")
     val stopEclairF = eclairRpcClient.stop()
     val killBitcoindF = BitcoindRpcTestUtil.stopServer(bitcoindRpc)
+    val iskilled = eclairRpcClient.isStopped
 
     for {
       _ <- killBitcoindF
       stopped <- stopEclairF
+      killed <- iskilled
     } yield {
-      if (stopped)
+      if (iskilled.isCompleted)
```

Comment:
why are you calling `.isCompleted` here? Since you do `killed <- iskilled` on the line above, that means that this future must be completed successfully to make it to this line of code.

Reply:
How should I get the debugger to throw a message in the case that it does not stop successfully? Just catch it with an exception instead?

Reply:
```scala
    val shutdownF = for {
      _ <- killBitcoindF
      stopped <- stopEclairF
      killed <- iskilled
    } yield {
      logger.debug(
        "Successfully shutdown eclair and it's corresponding bitcoind")
    }
    
    shutdownF.failed.foreach { err: Throwable =>
      //do whatever
      
    }
    
    shutdownF
```